### PR TITLE
TRON-1611 Handle KubernetesTask State Machine Transitions

### PR DESCRIPTION
**Pull request by lanceleih**
### Description
Tron now checks to see which state machine it will transition to with every new pod event. Pod events are based on the pod lifecycle phases.

### Test
`make test - PASSED`

### CODE REVIEW

File: tron/kubernetes.py [hunk 1]
```diff
@@ -123,7 +123,42 @@ def handle_event(self, event: Event) -> None:
...
+        if k8s_type == "pending":
+            pass
+        elif k8s_type == "running":
+            self.started()
+        elif k8s_type == "succeeded":
+            self.exited(0)
+        elif k8s_type == "failed":
+            self.exited(1)
+        elif k8s_type == "unknown":
+            self.log.warning("Kubernetes does not know anything about this task, it is UNKOWN")
```
Comment (jfongatyelp) [inline tron/kubernetes.py:135]: This is interesting, I feel like taskproc isn't submitting all of these `platform_type`s, we only seem to submit `task_event`s with:
* `killed`
* `finished`
* `FAILED` (uh oh, was this a typo?)
* `running`
* `unknown`

Are we sending the others (`pending`, `succeeded`, `failed` (I still think that's a typo)) from some other source?

Comment (lanceleih) [reply tron/kubernetes.py]: Nope. This was my mistake! I was misunderstood and was using pod phases instead of the ones emitted by TaskProc. Changed it.

**Review by jfongatyelp — APPROVED** [review]
I think this looks sane to me, though I'm not all that familiar with what these KubernetesTask methods actually do.  I think it makes sense to push this out so we can iterate!

File: tron/kubernetes.py [hunk 2]
```diff
@@ -123,7 +123,42 @@ def handle_event(self, event: Event) -> None:
...
+            self.log.warning(
+                "This can happen for any number of reasons, and Tron can't know if the task ran or not at all!"
+            )
+            self.log.warning("If you want Tron to RUN it (again) anyway, retry it with:")
+            self.log.warning(f"    tronctl retry {self.id}")
+            self.log.warning("If you want Tron to NOT run it and consider it as a success, skip it with:")
+            self.log.warning(f"    tronctl skip {self.id}")
+            self.log.warning("If you want Tron to NOT run it and consider it as a failure, fail it with:")
+            self.log.warning(f"    tronctl fail {self.id}")
+            self.exited(None)
```
[the remark this answers is not in the document]
Comment (lanceleih) [reply tron/kubernetes.py]: Modified it to be checking for "lost" instead of "unknown".

**Pull request merged**